packages/jbrowse-web/src/plugins/BigWigAdapter/BigWigAdapter.js

```diff
@@ -24,28 +39,83 @@ export default class BigWigAdapter extends BaseAdapter {
     return ((await this.bigwig.getHeader()).refsByNumber[refId] || {}).name
   }
 
-  async getGlobalStats() {
-    const header = await this.bigwig.getHeader()
+  async getGlobalStats(abortSignal) {
+    const header = await this.bigwig.getHeader(abortSignal)
     return header.totalSummary
   }
 
+  // todo: incorporate summary blocks
+  getRegionStats(region, abortSignal) {
+    const { refName, start, end, bpPerPx } = region
+    return this.statsCache.get(
+      `${refName}_${start}_${end}_${bpPerPx}`,
+      region,
+      abortSignal,
+    )
+  }
+
+  // todo: add caching
+  async getMultiRegionStats(regions = [], abortSignal) {
+    if (!regions.length) return undefined
+    const feats = await Promise.all(
+      regions.map(r => this.getRegionStats(r, abortSignal)),
+    )
+
+    const scoreMax = feats
+      .map(s => s.scoreMax)
+      .reduce((acc, curr) => Math.max(acc, curr))
+    const scoreMin = feats
+      .map(s => s.scoreMin)
+      .reduce((acc, curr) => Math.min(acc, curr))
+    const scoreSum = feats.map(s => s.scoreSum).reduce((a, b) => a + b, 0)
+    const scoreSumSquares = feats
+      .map(s => s.scoreSumSquares)
+      .reduce((a, b) => a + b, 0)
+    const featureCount = feats
+      .map(s => s.featureCount)
+      .reduce((a, b) => a + b, 0)
+    const basesCovered = feats
+      .map(s => s.basesCovered)
+      .reduce((a, b) => a + b, 0)
+
```

Comment:
while clean, the map-reduce constructions above are probably going to be rather slow.  here it's just one per region though, so not a big deal.

Reply:
I changed at least one other stats function to use direct for loops rather than these reduce statements. I'll will consider broadening this. Thanks though